.github/workflows/release-quil-rs.yml

```
@@ -0,0 +1,26 @@
+name: Release crates
+
+on:
+  release:
+    types: [published]
+    
+jobs:
+  is-quil-rs-release:
+    runs-on: ubuntu-latest
+    if: ${{ github.event_name == 'workflow_dispatch' || startswith(github.event.release.tag_name, 'quil-rs/v') }}
+    steps:
+      - run: echo "release tag starts with quil-rs/v, proceeding with release"
```

What the motivation/expected situation is to handle manual releases - wouldn't manually publishing a release tag give a similar effect? In a similar pyquil MR (see [comment](https://github.com/rigetti/pyquil/pull/1571/files#r1186326866)) I was hoping to align with the dual publish/manual event pattern, but didn't see a clean alternative to relying on `github.event.release.tag_name` in the manual case.

Also, is it intended that this have an `on: workflow_dispatch` section too, based on the `if` condition here?

```suggestion
on:
  release:
    types: [published]
  workflow_dispatch:
    description: "Manually publish release"
    
jobs:
  is-quil-rs-release:
    runs-on: ubuntu-latest
    if: ${{ github.event_name == 'workflow_dispatch' || startswith(github.event.release.tag_name, 'quil-rs/v') }}
    steps:
      - run: echo "release tag starts with quil-rs/v, proceeding with release"
```

---

Yeah, was intending to have a `workflow_dispatch` trigger here, good catch!

As for why it's there, it's mostly an escape hatch when something goes wrong with the release pipeline that is retry-able after knope has already committed version bumps, changelogs, and created the release tag. For example, a couple of weeks ago I released `qcs-sdk-python`, and [publishing to pypi failed on a 504](https://github.com/rigetti/qcs-sdk-rust/actions/runs/4928837074/jobs/8808162101#step:3:41). You are right in that recreating the release tag would work, but you would have to force delete the existing tag for the current release and recreate it, which seems more burdensome and error prone than the `workflow_dispatch` approach.

As for your use case, I think we could call the docker workflow at the end of a successful publish (see: [reusing workflows](https://docs.github.com/en/actions/using-workflows/reusing-workflows). That would cover the "happy path" and ensure the docker workflow ran after the package was published. Though, I think we'd also want a `workflow_dispatch` trigger on the docker workflow just in case the PyPI publish succeeds but the docker one fails on some transient error. PyPI returns an error if you try to publish an existing version, so you couldn't just re-trigger the Python release workflow to get the docker one to run after in that case. Assuming Docker also returns an error if you try to publish a duplicate tag, I don't see any harm in having the manual trigger. Does all that make sense? Hopefully I understood your question correctly.